### PR TITLE
Add predicate func for pod controller and node controller +1

### PR DESCRIPTION
**Pull request by zhengxiexie**
Update node predicate func, filter all the conditions, capacity and allocatable compare. Add pod predicate func, remove host network check in the beginning of reconcile.

### CODE REVIEW

**Review by yanjunz97 — DISMISSED** [review]
LGTM

**Comment by timdengyun** [comment]
` remove host network check in the beginning of reconcile.` @zhengxiexie 
 could you explain a bit why we need to do this?

**Comment by zhengxiexie** [comment]
> ` remove host network check in the beginning of reconcile.` @zhengxiexie could you explain a bit why we need to do this?

Yes, put it in predicate to avoid continuesly printing `pod is not scheduled on node yet, skipping`

File: pkg/controllers/pod/pod_controller.go [hunk 1]
```diff
@@ -270,3 +259,34 @@ func (r *PodReconciler) GetSubnetPathForPod(ctx context.Context, pod *v1.Pod) (s
...
+		if !ok {
+			return true
+		}
+
+		if pod.Spec.HostNetwork {
+			return false
+		}
+		return true
+	},
+	DeleteFunc: func(e event.DeleteEvent) bool { return false },
```
[the remark this answers is not in the document]
Comment (zhengxiexie) [reply pkg/controllers/pod/pod_controller.go]: ok

**Review by timdengyun — APPROVED** [review]
LGTM

**Review by timdengyun — APPROVED** [review]
LGTM

**Review by yanjunz97 — APPROVED** [review]

**Pull request merged**